py-polars/src/conversion.rs

```diff
@@ -346,7 +347,16 @@ impl ToPyObject for Wrap<DataType> {
             },
             #[cfg(feature = "object")]
             DataType::Object(_) => pl.getattr(intern!(py, "Object")).unwrap().into(),
-            DataType::Categorical(_) => pl.getattr(intern!(py, "Categorical")).unwrap().into(),
+            DataType::Categorical(rev_map) => {
+                if let Some(rev_map) = rev_map {
+                    if let RevMapping::Enum(categories, _) = &**rev_map {
```

Comment:
I'm not sure what the impact is of having multiple Python data types (`Categorical`, `Enum) represented as a single data type on the Rust side (`Categorical`). But I'll defer to Ritchie as he knows much more about that than me.